File: Examples2018/UIRatingsBar/app/src/main/java/course/examples/ui/ratingsbar/RatingsBarActivity.java
package course.examples.ui.ratingsbar;

import android.app.Activity;
import android.os.Bundle;
import android.widget.RatingBar;
import android.widget.RatingBar.OnRatingBarChangeListener;
import android.widget.TextView;


public class RatingsBarActivity extends Activity {

	@Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);

        final TextView tv = findViewById(R.id.textView);
        final RatingBar bar = findViewById(R.id.ratingbar);
      
        bar.setOnRatingBarChangeListener(new OnRatingBarChangeListener() {
	
        	// Called when the user swipes the RatingBar
        	@Override
			public void onRatingChanged(RatingBar ratingBar, float rating, boolean fromUser) {
                tv.setText(getString(R.string.rating_string, rating));
            }
		});
    }
}

File: ExamplesKotlin/UIRatingsBar/app/src/main/java/course/examples/ui/ratingsbar/RatingsBarActivity.kt
package course.examples.ui.ratingsbar

import android.app.Activity
import android.os.Bundle
import android.widget.RatingBar
import android.widget.RatingBar.OnRatingBarChangeListener
import android.widget.TextView


class RatingsBarActivity : Activity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        val tv = findViewById<TextView>(R.id.textView)

       findViewById<RatingBar>(R.id.ratingbar)?.onRatingBarChangeListener =
            OnRatingBarChangeListener { _, rating, _ ->
                // Called when the user swipes the RatingBar
                tv.text = getString(R.string.rating_string, rating)
            }
    }
}